boards/radxa-zero2/0001-radxa-zero2-board-enablement.patch

```diff
@@ -0,0 +1,690 @@
++CONFIG_SYS_WHITE_ON_BLACK=y
++CONFIG_VIDEO_MESON=y
++CONFIG_VIDEO_DT_SIMPLEFB=y
++CONFIG_SPLASH_SCREEN=y
++CONFIG_SPLASH_SCREEN_ALIGN=y
++# CONFIG_VIDEO_BMP_RLE8 is not set
++# CONFIG_BMP_16BPP is not set
++# CONFIG_BMP_24BPP is not set
++# CONFIG_BMP_32BPP is not set
++CONFIG_OF_LIBFDT_OVERLAY=y
+-- 
+2.34.0
+
+
+From caf7d9fb9d81c8d812673573f7a92fbab7747379 Mon Sep 17 00:00:00 2001
+From: Yuntian Zhang <yt@radxa.com>
+Date: Mon, 17 Jan 2022 14:11:14 +0800
+Subject: [PATCH 2/3] ARM: dts: add support for Radxa Zero 2
+
+Import the initial dts from RadxaYuntian/kernel/zero2-bringup repo.
+
+Signed-off-by: Yuntian Zhang <yt@radxa.com>
+(cherry picked from commit 28e43804e0f3ff399efc9b42ad5dbdf641923c1b)
+---
+ arch/arm/dts/Makefile                         |   1 +
+ .../dts/meson-g12b-radxa-zero2-u-boot.dtsi    |   7 +
+ arch/arm/dts/meson-g12b-radxa-zero2.dts       | 492 ++++++++++++++++++
+ 3 files changed, 500 insertions(+)
+ create mode 100644 arch/arm/dts/meson-g12b-radxa-zero2-u-boot.dtsi
+ create mode 100644 arch/arm/dts/meson-g12b-radxa-zero2.dts
+
+diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
+index fc16a57e60b..330d6b49bee 100644
+--- a/arch/arm/dts/Makefile
++++ b/arch/arm/dts/Makefile
+@@ -180,6 +180,7 @@ dtb-$(CONFIG_ARCH_MESON) += \
+ 	meson-g12b-odroid-n2.dtb \
+ 	meson-g12b-odroid-n2-plus.dtb \
+ 	meson-g12b-a311d-khadas-vim3.dtb \
++	meson-g12b-radxa-zero2.dtb \
+ 	meson-sm1-khadas-vim3l.dtb \
+ 	meson-sm1-odroid-c4.dtb \
+ 	meson-sm1-sei610.dtb
+diff --git a/arch/arm/dts/meson-g12b-radxa-zero2-u-boot.dtsi b/arch/arm/dts/meson-g12b-radxa-zero2-u-boot.dtsi
+new file mode 100644
+index 00000000000..236f2468dc2
+--- /dev/null
++++ b/arch/arm/dts/meson-g12b-radxa-zero2-u-boot.dtsi
+@@ -0,0 +1,7 @@
++// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
++/*
++ * Copyright (c) 2019 BayLibre, SAS.
++ * Author: Neil Armstrong <narmstrong@baylibre.com>
++ */
++
++#include "meson-g12-common-u-boot.dtsi"
+diff --git a/arch/arm/dts/meson-g12b-radxa-zero2.dts b/arch/arm/dts/meson-g12b-radxa-zero2.dts
+new file mode 100644
+index 00000000000..be7a1685da0
+--- /dev/null
++++ b/arch/arm/dts/meson-g12b-radxa-zero2.dts
+@@ -0,0 +1,492 @@
++// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
++/*
++ * Copyright (c) 2019 BayLibre, SAS
++ * Author: Neil Armstrong <narmstrong@baylibre.com>
++ * Copyright (c) 2019 Christian Hewitt <christianshewitt@gmail.com>
++ * Copyright (c) 2022 Radxa Limited
++ * Author: Yuntian Zhang <yt@radxa.com>
++ */
++
++/dts-v1/;
++
++#include "meson-g12b-a311d.dtsi"
++#include <dt-bindings/input/input.h>
++#include <dt-bindings/leds/common.h>
++#include <dt-bindings/gpio/meson-g12a-gpio.h>
++#include <dt-bindings/sound/meson-g12a-tohdmitx.h>
++
++/ {
++	compatible = "radxa,zero2", "amlogic,a311d", "amlogic,g12b";
++	model = "Radxa Zero2";
++
++	aliases {
++		serial0 = &uart_AO;
++	};
++
++	chosen {
++		stdout-path = "serial0:115200n8";
++	};
++
++	memory@0 {
++		device_type = "memory";
++		reg = <0x0 0x0 0x0 0x80000000>;
```

Comment:
`0x80000000` bytes is 2GiB, and indeed in the OS I see 2GiB available.

It seems there is no SKUs with less than 4GiB listed in the SKUs list. Shouldn't this be `0x100000000`?

*Z32M* on what I presume is the RAM chip makes me think that yes it's 4GiB.

Alternatively, I guess this could be detected during firmware startup, but it seems `meson` in U-Boot doesn't have facilities for this, or I'm not seeing it.

Reply:
That should be a 2GB chip. We actually have 1 board fitted with 4GB memory for Android, and when I tested the same dtb, U-Boot and Linux were able to detect the full size of memory. Rest of the boards are using 2GB memory since this is just some samples.

Also integers used there are 32 bits, so if you want to specify 4GB you need to do `reg = <0x0 0x0 0x1 0x0>;`.

Reply:
So Z32M-G 2052 here is not 32 megabits (`32 / 8 → 4`)? Wrong assumption on my end. I couldn't find specs, or even a sales sheet listing that product code. (Though I suspect 2052 could be a date code.)

With the testing you've done, assuming indeed the board I have is fitted with 2GiB, then there's no issue I figure.

I still wonder: should that node represent the amount of RAM the end-user product is expected to use?

Reply:
Fals alarm. Got it, tested things by reducing the hardcoded value to 1GiB, and yeah, everything is fin.

I guess what confused me was the combination of:

 - Not being aware that it had 2GiB instead of 4GiB of RAM
 - The *Z32M* marking which I assumed meant 32 gigabits
 - Coincidentally the value in the DT was 2GiB